### PR TITLE
feat(groq): autogenerated update

### PR DESCRIPTION
**Pull request by polsala**
## Rationale
- Automated proposal from the Groq generator delivering a fresh community utility.

## Why safe to merge
- Utility is isolated to \`utils/<util_name>\`.
- README + tests ship together (see folder contents).
- No secrets or credentials touched.

## Test Plan
- Follow the instructions in the generated README

## Links
- Generated docs and examples committed alongside this change.

## Mock Justification
- Not applicable; generator did not introduce new mocks.

### CODE REVIEW

**Comment by polsala** [comment]
Here is a consolidated review of the proposed changes:

✅ What’s solid
*   **Robust Core Logic:** The `sweep_dust_bunnies` function demonstrates a well-thought-out approach with two passes using `os.walk(topdown=False)` to correctly handle file removals before directory removals, ensuring that directories that become empty are also cleaned.
*   **Safe Defaults:** The `dry_run=True` default for `sweep_dust_bunnies` is an excellent safety measure, preventing accidental data loss when the utility is run without explicit removal intent.
*   **Comprehensive Error Handling:** The utility gracefully handles `OSError` in file system operations (`get_file_age_days`, `is_directory_empty`, `sweep_dust_bunnies`), preventing crashes and providing informative messages.
*   **Clear Reporting:** The returned dictionary from `sweep_dust_bunnies` provides a concise summary of found and removed items, which is valuable for user feedback and programmatic integration.

🧪 Tests
*   **Excellent Test Coverage:** The `test_sweeper.py` module provides thorough unit tests for all core functions, including edge cases like `OSError` during file system access and invalid paths.
*   **Deterministic Testing:** The use of `setUp` to define a `mock_current_time` and extensive patching of `time.time` ensures that age-based calculations are deterministic and not reliant on the actual system time.
*   **Realistic Mocking of File System:** The `side_effect` implementations for `os.walk`, `os.path.getmtime`, and `os.listdir` in the `dry_run` and `actual_run` tests create a highly realistic simulated file system environment, allowing for robust testing of the traversal and removal logic without touching the actual disk.

🔒 Security
*   **Path Traversal Mitigation:** The utility operates strictly within the `target_path` provided. There's no apparent mechanism for path traversal vulnerabilities (e.g., `../`) that would allow it to operate outside the intended scope.
*   **No Privilege Escalation:** The script uses standard file system operations (`os.remove`, `os.rmdir`) which respect the permissions of the user running the script. There are no `sudo` or other privilege escalation attempts.
*   **Dry Run Default:** The `dry_run=True` default significantly reduces the risk of accidental data loss, acting as a primary safety mechanism.
*   **Input Validation:** The initial check `if not os.path.isdir(target_path):` prevents the script from attempting operations on non-existent or invalid root paths, which could otherwise lead to unexpected behavior or errors.

🧩 Docs/DX
*   **Inconsistent Usage Path in README:** The `README.md` states "To run the sweeper, navigate to the `src` directory and execute the `sweeper.py` script," but then provides usage examples that assume execution from the *repository root* (e.g., `python3 utils/cosmic-dust-bunny-sweeper/src/sweeper.py`). This creates confusion regarding the correct execution context.
    *   **Actionable Feedback:** Standardize the usage instructions. Either consistently instruct users to run from the repository root (and adjust the `README` text accordingly), or provide examples that work when run from within the `src` directory (e.g., `python3 sweeper.py` after `cd .../src`).
    *   **Code Snippet Example (Option: Run from repo root):**
        ```diff
        --- a/utils/nightly-cosmic-dust-bunny-sweeper/utils/cosmic-dust-bunny-sweeper/README.md
        +++ b/utils/nightly-cosmic-dust-bunny-sweeper/utils/cosmic-dust-bunny-sweeper/README.md
        @@ -20,10 +20,10 @@
         ## Usage
         
         To run the sweeper, execute the `sweeper.py` script from the repository root.
         
         ```bash
        -python3 utils/cosmic-dust-bunny-sweeper/src/sweeper.py --path /path/to/clean --age-days 30
        +python3 utils/nightly-cosmic-dust-bunny-sweeper/utils/cosmic-dust-bunny-sweeper/src/sweeper.py --path /path/to/clean --age-days 30
         ```
         
         ### Arguments:
        ```
*   **Missing `if __name__ == "__main__":` block:** For a standalone Python script that uses `argparse`, it is best practice to encapsulate the main execution logic within an `if __name__ == "__main__":` block. This ensures that the code inside this block only runs when the script is executed directly, not when it's imported as a module.
    *   **Actionable Feedback:** Add a `main()` function to `sweeper.py` that handles argument parsing and calls `sweep_dust_bunnies`, and then invoke `main()` within an `if __name__ == "__main__":` block.
    *   **Code Snippet Example:**
        ```python
        # ... (imports and functions) ...

        def main():
            parser = argparse.ArgumentParser(description="Clean up old files and empty directories.")
            parser.add_argument('--path', required=True, help='The root directory to scan for dust bunnies.')
            parser.add_argument('--age-days', type=int, default=30, help='Files older than this many days will be considered for removal.')
            parser.add_argument('--dry-run', action='store_true', help='If present, only report what would be removed, without deleting anything.')
            parser.add_argument('--verbose', action='store_true', help='Print detailed information about each file/directory found.')
            args = parser.parse_args()

            print(f"Starting cosmic dust bunny sweep in '{args.path}' (dry_run={args.dry_run}, verbose={args.verbose})...")
            report = sweep_dust_bunnies(args.path, args.age_days, args.dry_run, args.verbose)

            print("\n--- Sweep Report ---")
            print(f"Files found older than {args.age_days} days: {len(report['found_files'])}")
            print(f"Files removed: {len(report['removed_files'])}")
            print(f"Empty directories found: {len(report['found_empty_dirs'])}")
            print(f"Empty directories removed: {len(report['removed_empty_dirs'])}")
            if args.dry_run:
                print("\n(This was a DRY RUN - no changes were made to the file system.)")

        if __name__ == "__main__":
            main()
        ```

🧱 Mocks/Fakes
*   **Clear Mock Justification:** The `test_sweeper.py` includes clear comments explaining the rationale behind patching `os` and `time` functions, which is excellent for maintainability and understanding.
*   **Effective Isolation:** The tests effectively isolate the `sweeper.py` logic from the actual file system and system time, ensuring tests are fast, reliable, and independent of the test environment.
*   **Comprehensive Mocking:** All external dependencies (`os.path.getmtime`, `os.listdir`, `os.walk`, `os.remove`, `os.rmdir`, `os.path.isdir`, `time.time`) are appropriately mocked, demonstrating a deep understanding of how to test file system interactions.
*   **Complex Mocking Scenarios:** The use of `side_effect` for `os.walk`, `os.path.getmtime`, and `os.listdir` to simulate complex file system structures and modification times is highly effective and demonstrates advanced mocking techniques.
